### PR TITLE
Clarify Tree.nodes

### LINKED ISSUE #1285: Clarify that Tree.nodes list nodes in traversal order
We don't really explain what we mean by "in a tree" either - only nodes that are reachable from a root will be listed.

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #1285

### CODE REVIEW

File: python/tskit/trees.py [hunk 1]
```diff
@@ -2136,10 +2136,16 @@ def _minlex_postorder_traversal(self, u):
 
     def nodes(self, root=None, order="preorder"):
         """
-        Returns an iterator over the node IDs in this tree. If the root parameter
-        is provided, iterate over the node IDs in the subtree rooted at this
-        node. If this is None, iterate over all node IDs. If the order parameter
-        is provided, iterate over the nodes in required tree traversal order.
+        Returns an iterator over the node IDs reachable from the root(s) in this tree. If
+        the root parameter is provided, only the subtree rooted at this single root node
+        will be iterated over. If this parameter is None, iterate over the node IDs in
```
Comment (jeromekelleher) [inline python/tskit/trees.py:2141]: Maybe link to the ``roots()`` method, or somewhere that there's a definition of what a root is?

File: python/tskit/trees.py [hunk 2]
```diff
@@ -2136,10 +2136,16 @@ def _minlex_postorder_traversal(self, u):
...
+        the subtrees rooted at each root node in turn. If the order parameter is
+        provided, iterate over the nodes in each subtree in the specified tree traversal
+        order.
+
+        .. note::
+            Not all nodes in the region spanned by this Tree may be
```
Comment (jeromekelleher) [inline python/tskit/trees.py:2147]: Not sure this helps - people should be comfortable with the idea that most nodes are not "in" any particular tree. The confusing thing is that we can have *topology* that's not reachable from a root (if we have non-sample leaves that don't connect up with a rooted subtree). This is pretty obscure though.

**Review by jeromekelleher — COMMENTED** [review]
LGTM, minor clarifications.

File: python/tskit/trees.py (hunk 1 above)
Comment (benjeffery) [reply python/tskit/trees.py:2142]: Done

File: python/tskit/trees.py (hunk 2 above)
Comment (benjeffery) [reply python/tskit/trees.py]: I've removed the note

**Comment by benjeffery** [comment]
@jeromekelleher Tweaked to comments.

**Review by jeromekelleher — APPROVED** [review]
LGTM

**Pull request merged**